core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -2036,6 +2036,7 @@ public function source()
 		if (Input::post('FORM_SUBMIT') == 'tl_files')
 		{
 			$strSource = System::getContainer()->get('request_stack')->getCurrentRequest()->request->get('source');
+			$strSource = str_replace("\r\n", "\n", $strSource); // see #5096
```

Comment:
In https://github.com/contao/contao/issues/5096#issuecomment-1353189856 we wrote _“we want detect and normalize accordingly”_ so I think we wanted to detect the line endings?

```suggestion
			$strEnding = preg_match_all('/(?<!\r)\n/', $strContent) >= preg_match_all('/\r\n/', $strContent) ? "\n" : "\r\n";
			$strSource = str_replace("\r\n", $strEnding, $strSource); // see #5096
```

Reply:
That would mean that my existing templates would never be fixed, right? 

Since browsers always send `\r\n` line endings, my existing custom templates now have them. And if we detect that, the method would incorrectly assume that I want to keep `\r\n`, even though I do not.

Reply:
I like the detection more, because this way when saving the template without making any changes the file is kept unchanged.

> Since browsers always send `\r\n` line endings, my existing custom templates now have them.

If this was not a problem for you, they can stay that way I think.

But I’m fine either way. What do you prefer @fritzmg ?

Reply:
imho normalizing to `\n` is fine, but I prefer auto-detection. We should definitely _not_ normalize to `PHP_EOL`.

Reply:
So we will normalize new templates but we will not provide an easy way to also fix existing ones?

Reply:
IMO “fix” is not the right word here. Contao is used as a text editor here, and most text editors (AFAIK) use the line endings that are already in use by existing files.